### ISSUE TITLE
Potential semantic change by Rails/FindEach cop

### ISSUE THREAD

**Issue by bootleq** [bbatsov/rubocop#2313]
Rails/FindEach cop suggests to change

```ruby
User.where(name: name).each { nil }
```

to

```ruby
User.where(name: name).find_each { nil }
```

which might cause unexpected semantic change:

1. `each {...}` returns Array, while `find_each {...}` returns nil in this case.
2. [`find_each`][find_each] does not respect `limit` and `order` conditions.

[find_each]: http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each

**Comment by alexdowad** [comment]
I think using `obj.each` if you actually want an `Array` is poor coding style. It's not clear that you are doing that to get an array.

However, if the other contributors disagree, I do have code which can "fix" this.

The point about `limit` and `order` is a good one.

**Comment by alexdowad** [comment]
Just opened a PR which fixes this point regarding `order` and `limit`. I read on an Internet forum that `find_each` also ignores `select(:field1, :field2, ...)`, so I included that too.

The point about the return value is (deliberately) not fixed.
